packages/core/http/core-http-versioned-router-server-internal/src/core_versioned_route.ts

```diff
@@ -155,19 +154,26 @@ export class CoreVersionedRoute implements VersionedRoute {
     return result;
   };
 
-  public addVersion(options: Options, handler: RequestHandler<any, any, any, any>): VersionedRoute {
-    if (this.handlers.has(options.version)) {
+  private validateVersion(version: string) {
+    const float = parseFloat(version);
+    if (!isFinite(float) || isNaN(float) || float <= 0 || Math.round(float) !== float) {
+      throw new Error(
+        `Invalid version number. Received "${version}", expected any finite, whole number greater than 0.`
+      );
+    }
```

Comment:
NIT: we may want to extract a `isValidRouteVersion` helper or something. We could easily imagine needing to use it elsewhere later.

Reply:
Good point!